### DISCUSSION TITLE
[Locked] ANDROID_NDK_ROOT environment variable should be preset

### DISCUSSION THREAD

**Discussion by leleliu008** [actions/virtual-environments#2399] [Installed software]
`Android NDK` has been pre-installed in virtual-environments, but not assign to a environment variable. `ANDROID_NDK_ROOT` is widely used by many other  softwares. so , it should be preset.

**Comment by maxim-lobanov** [comment]
@dmitry-shibanov could you please take a look? Any concerns to preset this variable?

**Reply by maxim-lobanov** [reply]
I guess we already have variables `ANDROID_NDK_HOME` and `ANDROID_NDK_PATH`

**Reply by leleliu008** [reply]
`macos-10.15` `macos-11.0` has preset `ANDROID_NDK_HOME="/Users/runner/Library/Android/sdk/ndk-bundle"` but `ubuntu` has not preset

**Reply by maxim-lobanov** [reply]
Yep, I see. `ANDROID_NDK_HOME` exists on macOS and Windows but missed on Ubuntu.
@leleliu008 does it work for you if we add `ANDROID_NDK_HOME` on Ubuntu instead of `ANDROID_NDK_ROOT` to be consistent with other images?

**Reply by leleliu008** [reply]
`ANDROID_NDK_HOME` and `ANDROID_NDK_ROOT` are widely used by many other softwares.  In most cases, If `ANDROID_NDK_HOME` is not defined, the value in `ANDROID_NDK_ROOT` is used. but some software only use `ANDROID_NDK_HOME`, others only use `ANDROID_NDK_ROOT`. so, both are preset is the best.

**Reply by leleliu008** [reply]
references:
1. https://github.com/openssl/openssl/issues/11205
2. https://github.com/android/ndk-samples

**Reply by dmitry-shibanov** [reply]
Hello this makes sense, we created an [issue](https://github.com/actions/virtual-environments/issues/2426) for this.

**Comment by AlenaSviridenko** [comment]
This has been done and is waiting for deployment, so I am locking discussion. Deployment progress can be tracked in https://github.com/actions/virtual-environments/issues/2426

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
